doc/source/configuration/cephadm.rst

```diff
@@ -308,6 +308,128 @@ should be used in the Kolla Manila configuration e.g.:
+  - "config set client.rgw rgw_keystone_token_cache_size '10000'"
+  - "config set client.rgw rgw_keystone_url https://{{ internal_net_vip_address }}:5000"
+  - "config set client.rgw rgw_keystone_verify_ssl false"
+  - "config set client.rgw rgw_max_attr_name_len '1000'"
+  - "config set client.rgw rgw_max_attr_size '1000'"
+  - "config set client.rgw rgw_max_attrs_num_in_req '1000'"
+  - "config set client.rgw rgw_s3_auth_use_keystone true"
+  - "config set client.rgw rgw_swift_account_in_url true"
+  - "config set client.rgw rgw_swift_versioning_enabled true"
+
+As we have configured Ceph to respond to Swift APIs, you will need to tell
+Kolla to account for this when registering Swift endpoints with Keystone. Also,
+when ``rgw_swift_account_in_url`` is set, the equivalent Kolla variable should
+be set in Kolla ``globals.yml`` too:
+
+.. code:: yaml
+
+  ceph_rgw_swift_compatibility: false
+  ceph_rgw_swift_account_in_url: true
+
+``secrets_ceph_rgw_keystone_password`` should be stored in the Kayobe
+``secrets.yml``, and set to the same value as ``ceph_rgw_keystone_password`` in
+the Kolla ``passwords.yml``. As such, you will need to configure Keystone
+before deploying the RADOS gateways. If you are using the Kolla load balancer
+(see :ref:`RGWs-with-hyper-converged-Ceph` for more info), you can specify the
+``haproxy`` and ``loadbalancer`` tags here too.
+
+.. code:: yaml
+
+  kayobe overcloud service deploy -kt ceph-rgw,keystone,haproxy,loadbalancer
+
+
+.. _RGWs-with-hyper-converged-Ceph:
+
+RGWs with hyper-converged Ceph
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you are using a hyper-converged Ceph setup (i.e. your OpenStack controllers
+and Ceph storage nodes share the same hosts), you will need to change the
+``rgw_frontend_port``. This is because port 80 (and 443) will be bound to the
+Kolla-deployed haproxy. You should choose a custom port that does not conflict
+with any OpenStack endpoints (``openstack endpoint list``), such as port 8100.
+
+You may also want to use the Kolla-deployed haproxy to load balance your RGWs.
+This means you will not need to define any Ceph ingress services. Instead, you
+add definitions of your Ceph hosts to Kolla ``globals.yml``:
+
+.. code:: yaml
+
+  ceph_rgw_hosts:
+  - host: controller1
+    ip: <host IP on storage net>
+    port: 8100
+  - host: controller2
+    ip: <host IP on storage net>
+    port: 8100
+  - host: controller3
+    ip: <host IP on storage net>
+    port: 8100
+
+HA with Ingress services
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ingress services are defined with the following. ``id`` should match name (not
+id) of the RGW service to which ingress will point to. ``spec`` is a service
+specification required by Cephadm to deploy the ingress (haproxy + keepalived
+pair).
+
+.. code:: yaml
+
+  cephadm_ingress_services:
+      - id: rgw.myrgw
+        spec:
+          frontend_port: 443
+          monitor_port: 1967
+          virtual_ip: 10.66.0.1/24
```

Comment:
ping @cityofships

Reply:
Sorry both, I missed your pings in the notifications flood somehow;

@markgoddard Definitely different from Kolla's VIP - unless you'd like keepaliveds fighting for that VIP. It's either Kolla making for an object store front end (HAproxy + keepalived) or Ceph's Ingress - e.g. for standalone clusters. Question on the VIP subnet - that's entirely up for the network design. It can be arbitrary really, outside of any Ceph's networks.

Reply:
I've written a bit to clarify this in the docs